vispy/geometry/generation.py

```diff
@@ -89,6 +89,250 @@ def create_cube():
     return vertices, filled, outline
 
 
+def create_plane(width=1,
+                 height=1,
+                 width_segments=1,
+                 height_segments=1,
+                 direction='+z'):
+    """ Generate vertices & indices for a filled and outlined plane.
+
+    Parameters
+    ----------
+    width : numeric
+        Plane width.
+    height : numeric
+        Plane height.
+    width_segments : int
+        Plane segments count along the width.
+    height_segments : numeric
+        Plane segments count along the height.
+    direction: unicode
+        ``{'-x', '+x', '-y', '+y', '-z', '+z'}``
+        Direction the plane will be facing.
+
+    Returns
+    -------
+    vertices : array
+        Array of vertices suitable for use as a VertexBuffer.
+    faces : array
+        Indices to use to produce a filled plane.
+    outline : array
+        Indices to use to produce an outline of the plane.
+
+    References
+    ----------
+    [1] Cabello, R. (n.d.). PlaneBufferGeometry.js. Retrieved May 12, 2015,
+        from https://github.com/mrdoob/three.js/blob/master/src/extras/geometries/PlaneBufferGeometry.js  #noqa
+    """
+
+    x_grid = width_segments
+    y_grid = height_segments
+
+    x_grid1 = x_grid + 1
+    y_grid1 = y_grid + 1
+
+    # Positions, normals and texcoords.
+    positions = np.zeros(x_grid1 * y_grid1 * 3)
+    normals = np.zeros(x_grid1 * y_grid1 * 3)
+    texcoords = np.zeros(x_grid1 * y_grid1 * 2)
+
+    offset = offset1 = 0
+    for i_y in range(y_grid1):
+        y = i_y * height / y_grid - height / 2
+        for i_x in range(x_grid1):
+            x = i_x * width / x_grid - width / 2
+
+            positions[offset] = x
+            positions[offset + 1] = - y
+
+            normals[offset + 2] = 1
+
+            texcoords[offset1] = i_x / x_grid
+            texcoords[offset1 + 1] = 1 - (i_y / y_grid)
+
+            offset += 3
+            offset1 += 2
+
+    # Faces and outline.
+    faces = np.zeros(x_grid * y_grid * 6, dtype=np.uint32)
+    outline = np.zeros(x_grid * y_grid * 8, dtype=np.uint32)
+
+    offset = offset1 = 0
+    for i_y in range(y_grid):
+        for i_x in range(x_grid):
+            a = i_x + x_grid1 * i_y
+            b = i_x + x_grid1 * (i_y + 1)
+            c = (i_x + 1) + x_grid1 * (i_y + 1)
+            d = (i_x + 1) + x_grid1 * i_y
+
+            faces[offset] = a
+            faces[offset + 1] = b
+            faces[offset + 2] = d
+
+            faces[offset + 3] = b
+            faces[offset + 4] = c
+            faces[offset + 5] = d
+
+            outline[offset1] = a
+            outline[offset1 + 1] = b
+
+            outline[offset1 + 2] = b
+            outline[offset1 + 3] = c
+
+            outline[offset1 + 4] = c
+            outline[offset1 + 5] = d
+
+            outline[offset1 + 6] = d
+            outline[offset1 + 7] = a
+
+            offset += 6
+            offset1 += 8
+
+    positions = np.reshape(positions, (-1, 3))
```

Comment:
Wait, if you were going to do this anyway, why not just allocate it with that shape in the first place instead of messing around with offsets?